src/v/archival/ntp_archiver_service.cc

```diff
@@ -230,9 +230,8 @@ ntp_archiver::upload_segment(upload_candidate candidate) {
     vlog(ctxlog.debug, "Uploading segment {} to {}", candidate, path);
 
     auto reset_func = [this, candidate] {
-        auto stream = candidate.source->reader().data_stream(
+        return candidate.source->reader().data_stream(
           candidate.file_offset, candidate.final_file_offset, _io_priority);
-        return stream;
```

Comment:
i think i haven't gotten to the part of the PR or this commit which can tell me what type is being returned. i'm guessing it is returning a future (because above we were co_awaiting on data_stream(..)) but then was this lambda not returning a future before?